notebooks/tuebingen_causality_pairs/tuebingen_pairs_notebook_v2.ipynb

```diff
@@ -1,1060 +1,421 @@
-    "            av_correct_ab += 1\n",
-    "\n",
-    "        if llm_output[(id, 0.3, i+1)]['llm_ba'] == 1 and saved_pairs_info[id]['ground_truth'] == \" L\":\n",
-    "            av_correct_ba += 1\n",
-    "        elif llm_output[(id, 0.3, i+1)]['llm_ba'] == 0 and saved_pairs_info[id]['ground_truth'] == \" R\":\n",
-    "            av_correct_ba += 1\n",
-    "\n",
-    "    av_correct_ab /= 5.0\n",
-    "    av_correct_ba /= 5.0\n",
-    "    \n",
-    "    temp : Dict = {}\n",
-    "\n",
-    "    temp['PairID'] = id\n",
-    "    temp['CorrectACauseB'] = av_correct_ab\n",
-    "    temp['CorrectBCauseA'] = av_correct_ba\n",
-    "    temp['VarA'] = saved_pairs_info[id]['var1']\n",
-    "    temp['VarB'] = saved_pairs_info[id]['var2'] \n",
-    "    temp['GroundTruth'] = saved_pairs_info[id]['ground_truth']\n",
-    "\n",
-    "    results[id] = temp\n",
-    "    print(results[id])\n",
-    "\n",
-    "\n"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "metadata": {},
-   "source": [
-    "#### Save to csv file"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 53,
-   "metadata": {},
-   "outputs": [
+      "cell_type": "code",
+      "source": [
+        "pip install guidance"
+      ],
+      "metadata": {
+        "id": "DMnZMAMnP1Oc"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "pip install python-dotenv"
+      ],
+      "metadata": {
+        "id": "zQ8BeAeuP1iv"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
     {
-     "name": "stdout",
-     "output_type": "stream",
-     "text": [
-      "CSV file 'gpt-4_results_straight_prompt_w_descriptions.csv' has been created.\n"
-     ]
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7adKIAb_MVED"
+      },
+      "outputs": [],
+      "source": [
+        "from dotenv import load_dotenv\n",
+        "from typing import Dict, List, Tuple\n",
+        "import guidance\n",
+        "import os\n",
+        "\n",
+        "load_dotenv()\n",
+        "\n",
+        "os.environ[\"OPENAI_API_KEY\"] = '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'"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "m9dRuOYgMVEF"
+      },
+      "outputs": [],
+      "source": [
+        "from pywhyllm.suggesters.tuebingen_model_suggester import TuebingenModelSuggester, Strategy\n",
+        "modeler = TuebingenModelSuggester('gpt-4')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "uTiIMdPIMVEF"
+      },
+      "outputs": [],
+      "source": [
+        "# TODO download the dataset if needed and then read in the CSV\n",
```

Comment:
Won't it be easier to read the dataset from a csv? You may want to create a csv yourself and add it to this folder.
That way, the code is more maintainable and any changes to the dataset can be easily done in the future.